io.c

```diff
@@ -5637,13 +5637,31 @@ rb_io_close(VALUE io)
  *  call-seq:
  *    close -> nil
  *
- *  Closes the stream, if it is open, after flushing any buffered writes
- *  to the operating system; does nothing if the stream is already closed.
- *  A stream is automatically closed when claimed by the garbage collector.
+ *  Closes the stream for both reading and writing
+ *  if open for either or both; returns +nil+.
  *
- *  If the stream was opened by IO.popen, #close sets global variable <tt>$?</tt>.
+ *  If the stream is open for writing, flushes any buffered writes
+ *  to the operating system before closing.
  *
- *  See also {Open and Closed Streams}[rdoc-ref:io_streams.rdoc@Open+and+Closed+Streams].
+ *  If the stream was opened by IO.popen, sets global variable <tt>$?</tt>
+ *  (child exit status).
+ *
+ *  Example:
+ *
+ *    IO.popen('ruby', 'r+') do |pipe|
+ *      STDOUT.puts pipe.closed?
+ *      pipe.close
+ *      STDOUT.puts $?
+ *      STDOUT.puts pipe.closed?
+ *    end
```

Comment:
Do we need to explicitly state `STDOUT` here? `puts` will default to `STDOUT` and these examples works without explicitly stating `STDOUT`.

Reply:
FYI: `puts` uses `$stdout` not `STDOUT` by default.

BTW, it seems that we can omit `STDOUT.` here.

Reply:
As I said to @peterzhu2118 offline earlier, doing IO.popen in irb on my Windows machine fouls $stdin and $stdout -- no output to irb after exiting the popen block.  So in irb, my examples need to do STDOUT.

I'll try these examples as free Ruby scripts, which I hope will not need STDOUT and will otherwise behave.